### PR TITLE
Fix scan find

### PR DESCRIPTION
**Pull request by simsong**

### CODE REVIEW

File: etc/CONFIGURE_MACOS.bash [hunk 1]
```diff
@@ -22,7 +22,7 @@ read
 
 # Note: openssl no longer required
 # Apple's provided flex is 2.6.4, which is the same that is provided by brew
-PKGS+="wget libtool autoconf automake libtool libxml2 libewf json-c"
+PKGS+="wget libtool autoconf automake libtool libxml2 libewf json-c re2 abseil pkg-config"
```
Comment (jonstewart) [inline etc/CONFIGURE_MACOS.bash:25]: If you're requiring `pkg-config`, you can add `ax_pkg_check_modules.m4` to your `m4` directory and then use `PKG_CHECK_MODULES()` in `configure.ac` to check on the existence/presence of pkg-config-managed packages and gather up their different CFLAGS and LIBS directives into namespaced variables, so you can incorporate those into `Makefile.am`.

See https://autotools.info/pkgconfig/pkg_check_modules.html

File: src/scan_find.cpp [hunk 2]
```diff
@@ -43,6 +46,41 @@ namespace {
     }
 }
 
+void scan_find_sbuf(scanner_params &sp, sbuf_t &sbuf)
+{
+    feature_recorder &f = sp.named_feature_recorder("find");
+
+    auto *tbuf = sbuf_t::sbuf_malloc(sp.sbuf->pos0, sp.sbuf->bufsize+1, sp.sbuf->bufsize+1);
+    memcpy(tbuf->malloc_buf(), sp.sbuf->get_buf(), sp.sbuf->bufsize);
```
Comment (jonstewart) [inline src/scan_find.cpp:54]: are you doing the `memcpy` solely so you can null-terminate it?

File: src/scan_find.cpp [hunk 3]
```diff
@@ -79,33 +117,16 @@ void scan_find(scanner_params &sp)
         /* The current regex library treats \0 as the end of a string.
          * So we make a copy of the current buffer to search that's one bigger, and the copy has a \0 at the end.
          * This is super-wasteful. Does Lightgrep have this problem?
```
Comment (jonstewart) [inline src/scan_find.cpp:119]: to answer definitively: no

**Review by jonstewart — COMMENTED** [review]
is usage of re2 actually in `be20_api`?

File: etc/CONFIGURE_MACOS.bash (hunk 1 above)
Comment (simsong) [reply etc/CONFIGURE_MACOS.bash:25]: Yes. I'm in fact using this now. See 
https://github.com/simsong/be20_api/blob/0bd6a2a8a72dffcc867cef478a3f84cf306f131f/be20_configure.m4#L31-L40

And you know what? I just spent 8 hours trying to debug this.

File: src/scan_find.cpp (hunk 2 above)
Comment (simsong) [reply src/scan_find.cpp:54]: Yes.

**Pull request merged**